iceoryx_posh/test/moduletests/test_posh_runtime.cpp

```diff
@@ -545,6 +546,276 @@ TEST_F(PoshRuntime_test, GetMiddlewareSubscriberWithQueueFullPolicySetToBlockPub
+    EXPECT_EQ(serverPort->m_chunkSenderData.m_memoryInfo.memoryType, portConfig.memoryInfo.memoryType);
+}
+
+TEST_F(PoshRuntime_test, GetMiddlewareServerWithQueueGreaterMaxCapacityClampsQueueToMaximum)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "91b21e80-0f98-4ae3-982c-54deaab93d96");
+    constexpr uint64_t MAX_QUEUE_CAPACITY = iox::popo::ServerChunkQueueConfig::MAX_QUEUE_CAPACITY;
+    const iox::capro::ServiceDescription sd{"819", "1", "20"};
+    iox::popo::ServerOptions serverOptions;
+    serverOptions.requestQueueCapacity = MAX_QUEUE_CAPACITY + 1U;
+
+    auto serverPort = m_runtime->getMiddlewareServer(sd, serverOptions);
+
+    ASSERT_THAT(serverPort, Ne(nullptr));
+    EXPECT_EQ(serverPort->m_chunkReceiverData.m_queue.capacity(), MAX_QUEUE_CAPACITY);
+}
+
+TEST_F(PoshRuntime_test, GetMiddlewareServerWithQueueCapacityZeroClampsQueueCapacityTo1)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "a28a30eb-f3be-43c9-a948-26c71c5f12c9");
+    const iox::capro::ServiceDescription sd{"89", "19", "20"};
+    iox::popo::ServerOptions serverOptions;
+    serverOptions.requestQueueCapacity = 0U;
+
+    auto serverPort = m_runtime->getMiddlewareServer(sd, serverOptions);
+
+    ASSERT_THAT(serverPort, Ne(nullptr));
+    EXPECT_EQ(serverPort->m_chunkReceiverData.m_queue.capacity(), 1U);
+}
+
+TEST_F(PoshRuntime_test, GetMiddlewareServerWhenMaxServerAreUsedResultsInServerlistOverflow)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "8f679838-3332-440c-aa95-d5c82d53a7cd");
+    auto serverOverflowDetected{false};
+    auto errorHandlerGuard = iox::ErrorHandler::setTemporaryErrorHandler(
+        [&](const iox::Error error, const std::function<void()>, const iox::ErrorLevel) {
+            if (error == iox::Error::kPORT_POOL__SERVERLIST_OVERFLOW)
+            {
+                serverOverflowDetected = true;
+            }
+        });
+
+    uint32_t i{0U};
+    for (; i < iox::MAX_SERVERS; ++i)
+    {
+        auto serverPort = m_runtime->getMiddlewareServer(
+            iox::capro::ServiceDescription(iox::capro::IdString_t(TruncateToCapacity, convert::toString(i)),
+                                           iox::capro::IdString_t(TruncateToCapacity, convert::toString(i + 1U)),
+                                           iox::capro::IdString_t(TruncateToCapacity, convert::toString(i + 2U))));
+        ASSERT_THAT(serverPort, Ne(nullptr));
+    }
+    EXPECT_FALSE(serverOverflowDetected);
+
+    auto serverPort = m_runtime->getMiddlewareServer(
+        iox::capro::ServiceDescription(iox::capro::IdString_t(TruncateToCapacity, convert::toString(i)),
+                                       iox::capro::IdString_t(TruncateToCapacity, convert::toString(i + 1U)),
+                                       iox::capro::IdString_t(TruncateToCapacity, convert::toString(i + 2U))));
+    EXPECT_THAT(serverPort, Eq(nullptr));
+    EXPECT_TRUE(serverOverflowDetected);
+}
+
+TEST_F(PoshRuntime_test, GetMiddlewareServerWithInvalidNameLeadsToTermination)
```

Comment:
```suggestion
TEST_F(PoshRuntime_test, GetMiddlewareServerWithInvalidNameLeadsToError)
```

Reply:
Can you please write `InvalidNodeName` instead of just `InvalidName`.